### ISSUE TITLE
Formatter: Don't reorder parameters in function calls 

### ISSUE THREAD

**Issue by konstin** [astral-sh/ruff#6498]
Currently, we format
```python
f(a=b, *args, **kwargs)
```
as 
```python
f(*args, a=b, **kwargs)
```
Instead, we should preserve the function order.

**Comment by NeilGirdhar** [comment]
Why do you think that's better?  The original order was contentious when we implemented PEP 448.  And there's one good reason to do the reorder:
```python
def g():
    print("g")
    return range(2)
def h():
    print("h")
    return 1

def f(*args, **kwargs):
    pass

f(a=h(), *g(), **{})  # Prints g then h!
```
Positional arguments are evaluated first.
``

**Comment by charliermarsh** [comment]
I believe we do warn when keyword arguments are provided in such an order: https://beta.ruff.rs/docs/rules/star-arg-unpacking-after-keyword-arg/ (but it's unconditional)

**Comment by NeilGirdhar** [comment]
@charliermarsh My mistake, I tested it but forgot to enable all the rules 😄 

@MichaReiser Fair enough, sorry for the noise!

**Comment by charliermarsh** [comment]
@NeilGirdhar - You're telling me you haven't memorized every single rule in Ruff?

**Comment by charliermarsh** [comment]
I honestly didn't realize that the evaluation order was independent of the literal order. I should update that rule documentation to include this, it's interesting.

**Comment by NeilGirdhar** [comment]
Good idea.

It's one of the few places I think.  Really glad to see you have rule B026.  Definitely an important one for preventing bugs.

Do you have statistics on how often your various rules are hit in public code?  If this rule is hit very rarely, we may be able to convince the Python people to syntax-warn, and eventually deprecate this form.
